www/js/app.js

```diff
@@ -886,6 +899,49 @@ function initServiceWorkerMessaging () {
     }
 }
 
+/**
+ * Function that handles a message of the messageChannel.
+ * This function will deal with the messages if useLibzim is set to true
+ *
+ * @param {Event} event The event object of the message channel
+ */
+async function handleMessageChannelByLibzim (event) {
+    // We received a message from the ServiceWorker
+    // The ServiceWorker asks for some content
+    const title = event.data.title;
+    const messagePort = event.ports[0];
+    try {
+        const ret = await selectedArchive.callLibzimWorker({ action: 'getEntryByPath', path: title })
+        if (ret === null) {
+            console.error('Title ' + title + ' not found in archive.');
+            messagePort.postMessage({ action: 'giveContent', title: title, content: '' });
+            return;
+        }
+
+        if (ret.mimetype === 'unknown') {
+            // We have a redirect to follow
+            // this is still a bit flawed, as we do not check if the redirect or the file doesnt exist
```

Comment:
I would say ""if the redirect or the file actually exist". In any case, `doesn't` needs and apostrophe if you keep it.